vm/src/stdlib/itertools.rs

```diff
@@ -1959,18 +1959,25 @@ mod decl {
 
     impl IterNext for PyItertoolsPairwise {
         fn next(zelf: &Py<Self>, vm: &VirtualMachine) -> PyResult<PyIterReturn> {
-            let old = match zelf.old.read().clone() {
+            let old_guard = {
```

Comment:
This is not a clone of old_guard, but clone of old, right?
```suggestion
            let old = {
```
or

```suggestion
            let old_cloned = {
```